### PR TITLE
Changed flop formulas for flop counter to take in shapes directly

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96581
* __->__ #96565

### CODE REVIEW

File: torch/utils/flop_counter.py [hunk 1]
```diff
@@ -91,58 +83,96 @@ def conv_flop_count(
...
-    query_shape, key_shape, value_shape, *_ = input_shapes
     return sdpa_flop_count(query_shape, key_shape)
 
+
+def sdpa_backward_flop_count(grad_out_shape, query_shape, key_shape):
+    total_flops = 0
+    b, h, s, d = query_shape
+    _b2, _h2, s2, _d2 = key_shape
+    _b3, _h3, _s3, _d3 = grad_out_shape
+    assert b == _b2 == _b3 and h == _h2 == _h3 and d == _d2 == _d3
```
Comment (drisspg) [inline torch/utils/flop_counter.py:149]: V can have a different embed dim from q,k and mem efficient supports that

**Review by zdevito — APPROVED** [review]
Formatting changes look good. I didn't check the sdpa formula.

File: torch/utils/flop_counter.py (hunk 1 above)
Comment (Chillee) [reply torch/utils/flop_counter.py:149]: Oh ok.

**Pull request closed without merging**